src/main/java/com/g4s8/ghman/bot/TkThread.java

```diff
@@ -75,23 +83,46 @@ public TmResponse act(final Update update) throws IOException {
                     issue.repo().coordinates(),
                     issue.number()
                 ).toString(),
-                new Issue.Smart(issue).title(),
+                smart.title(),
                 new Joined(
                     new TextOf("\n\n"),
                     new Mapped<>(
                         cmt -> new FormattedText(
-                        "[@%s](https://github.com/%s) > %s",
-                        cmt.author().login(),
-                        cmt.author().login(),
-                        cmt.body()
+                            "[@%s](https://github.com/%s) > %s",
+                            cmt.author().login(),
+                            cmt.author().login(),
+                            cmt.body()
                         ),
                         new Mapped<>(
                             Comment.Smart::new,
-                            issue.comments().iterate(new Date(thread.lastRead().toEpochMilli()))
+                            issue.comments().iterate(
+                                new Date(thread.lastRead().toEpochMilli())
+                            )
                         )
                     )
                 )
             )
         );
+        return new Unchecked<>(
+            new Ternary<>(
+                () -> smart.isOpen()
+                    && smart.author().equals(user.github().users().self()),
+                new RsInlineKeyboard(
+                    text,
+                    new IterableOf<>(
+                        new IterableOf<>(
+                            new MapEntry<>(
+                                "close",
+                                new FormattedText(
+                                    "click:notification.close?repo=%s&issue=%d",
+                                    issue.repo().coordinates(), issue.number()
+                                ).toString()
```

Comment:
@olenagerasimova please use `asString()` as it is the `Text` interface explicitly exposed behaviour